### PR TITLE
selector streamlining

### PR DESCRIPTION
**Pull request by adecaro**
This PR does the following:
- It introduces an integration test used to test the selectors
- It deprecates mailman (long glory to it)

### CODE REVIEW

File: docs/core-token.md [hunk 1]
```diff
@@ -12,9 +12,10 @@ token:
   enabled: true
 
   # token selector configuration allows to use different implementations of the token selector
-  # the "default" driver is the mailman implementation, other possible configurations are: "simple"
+  # the "sherdlock" driver is the mailman implementation, other possible configurations are: "simple"
```
Comment (mbrandenburger) [inline docs/core-token.md:15]:
```suggestion
  # the "sherdlock" driver is the default implementation, other possible configurations are: "simple"
```

File: integration/nwo/token/topology.go [hunk 2]
```diff
@@ -31,7 +31,8 @@ type Topology struct {
 	TopologyName string `yaml:"name,omitempty"`
 	TopologyType string `yaml:"type,omitempty"`
 
-	TMSs []*topology.TMS
+	Selector string
```
Comment (mbrandenburger) [inline integration/nwo/token/topology.go:34]: for consistency this should also be named `TokenSelector`

File: integration/token/fungible/dlog/dlog_test.go [hunk 3]
```diff
@@ -20,6 +20,7 @@ import (
 	"github.com/hyperledger-labs/fabric-token-sdk/integration/token/fungible"
 	"github.com/hyperledger-labs/fabric-token-sdk/integration/token/fungible/topology"
 	"github.com/hyperledger-labs/fabric-token-sdk/token/core/zkatdlog/crypto"
+	"github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/ginkgo/v2"
```
Comment (mbrandenburger) [inline integration/token/fungible/dlog/dlog_test.go:24]: double import

File: integration/token/fungible/dlog/dlog_test.go [hunk 4]
```diff
@@ -119,8 +129,10 @@ func newTestSuite(commType fsc.P2PCommunicationType, mask int, factor int, names
 			HSM:             mask&HSM > 0,
 			WebEnabled:      mask&WebEnabled > 0,
 			SDKs:            []api.SDK{&fdlog.SDK{}},
-			Monitoring:      true,
+			Monitoring:      false,
```
Comment (mbrandenburger) [inline integration/token/fungible/dlog/dlog_test.go:132]: are you sure this is part of the PR?

File: integration/ports.go [hunk 5]
```diff
@@ -24,29 +24,37 @@ const (
 	portsPerSuite = 10 * portsPerNode
 )
 
+var (
+	TokenSelectors = []string{"simple", "sherdlock"}
```
Comment (alexandrosfilios) [inline integration/ports.go:28]: Maybe we could have a constant for those

File: token/services/selector/simple/service.go [hunk 6]
```diff
@@ -19,6 +19,11 @@ import (
 
 var logger = logging.MustGetLogger("token-sdk.selector.simple")
 
+const (
```
Comment (alexandrosfilios) [inline token/services/selector/simple/service.go:22]: Do you think it makes sense to put these in a config?

**Review by alexandrosfilios — APPROVED** [review]
LGTM modulo comments

**Pull request merged**